### ISSUE TITLE
need help with python code 

### ISSUE THREAD

**Issue by muha99** [PrachiSinghai1105/Superstore#5]
hey guys i have created this code to predict if  machine needs of maintenance , so the code takes a data that containes some informations and makes a link between these features with the remainin guseful life of each machine which means the machine need of maintenance or not , so the problem is that i am not sure of this , i think i made somelogical mistakes in the script i everytime i try to test it with a new data without the remaining useful life column of course which is wht i want from the program topredict it gaves me an error can you guys help me please 
import pandas as pd
import csv
import joblib
from sklearn.metrics import accuracy_score, precision_score, recall_score, f1_score, roc_auc_score
import seaborn as sns
import matplotlib.pyplot as plt
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LogisticRegression
import pandas as pd
import csv
# Define the column names and data separately
column_names = ["Machine_ID","Timestamp","Machine_Temperature","Pressure","Humidity","Vibration","RemainingUsefulLife"]
machine = [
    (1, "2023-7-1 12:0:0", 140, 80, 40, 0.5, 210),
    (2, "2023-7-1 1:0:0", 135, 85, 42, 0.6, 190),
    (3, "2023-7-1 2:0:0", 142, 90, 38, 0.4, 175),
    (4, "2023-7-1 3:0:0", 150, 78, 45, 0.7, 160),
    (5, "2023-7-1 4:0:0", 132, 88, 41, 0.5, 145),
    (6, "2023-7-1 5:0:0", 137, 82, 43, 0.6, 135),
    (7, "2023-7-1 6:0:0", 145, 86, 39, 0.4, 120),
    (8, "2023-7-1 7:0:0", 139, 84, 44, 0.7, 100),
    (9, "2023-7-1 8:0:0", 128, 91, 37, 0.5, 85),
    (10, "2023-7-1 9:0:0", 133, 77, 40, 0.6, 70),
    (11, "2023-7-1 10:0:0", 148, 83, 41, 0.5, 60),
    (12, "2023-7-1 11:0:0", 141, 89, 36, 0.4, 45),
    (13, "2023-7-1 12:0:0", 130, 80, 39, 0.6, 30),
    (14, "2023-7-1 13:0:0", 136, 85, 43, 0.5, 20),
    (15, "2023-7-1 14:0:0", 143, 76, 42, 0.7, 10),
    (16, "2023-7-1 15:0:0", 129, 92, 38, 0.4, 5),
]
# Write the data to a CSV file
with open("machine.csv", "w", newline='') as machinedata:
    writer = csv.writer(machinedata)
    writer.writerow(column_names)  # Write column names
    writer.writerows(machine)  # Write data rows
# load the data
data = pd.read_csv('machine.csv')
#supprimer les informations inutiles
data=data.drop(['Machine_ID','Timestamp'],axis=1)
#remplir les faleur manquantes avec des 0
data=data.fillna(0)
#créer une valeur cible
data['MaintenanceNeeded']=(data['RemainingUsefulLife']<=30).astype(int)
# Diviser les données en fonctionnalités (X) et variable cible (y)
X = data.drop('MaintenanceNeeded', axis=1)
y = data['MaintenanceNeeded']
# Visualiser la distribution des données de maintenance
sns.countplot(x='MaintenanceNeeded', data=data)
plt.show()

# Visualiser les corrélations entre les caractéristiques et la maintenance
correlations = data.corr()
sns.heatmap(correlations, annot=True)
plt.show()
# Diviser les données en ensembles d'entraînement et de test
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

# Créer un modèle de régression logistique
model = LogisticRegression()
# Entraîner le modèle
model.fit(X_train, y_train)
# Prédire les classes
y_pred = model.predict(X_test)
# Calculer les métriques
accuracy = accuracy_score(y_test, y_pred)
precision = precision_score(y_test, y_pred)
recall = recall_score(y_test, y_pred)
f1 = f1_score(y_test, y_pred)
roc_auc = roc_auc_score(y_test, model.predict_proba(X_test)[:,1])
print(f'Accuracy: {accuracy}')
print(f'Precision: {precision}')
print(f'Recall: {recall}')
print(f'F1 Score: {f1}')
print(f'ROC AUC: {roc_auc}')
# Save the trained model to a file
joblib.dump(model,'machinemodel.pkl')

this is the code above  and this is the test code
 import pandas as pd
import joblib

# Define the column names for the new data (excluding "Machine_ID" and "Timestamp")
column_names = ["Machine_Temperature", "Pressure", "Humidity", "Vibration"]

# Define the new data
new_data = [
    (140, 80, 40, 0.5),
    (135, 85, 42, 0.6),
    (142, 90, 38, 0.4),
    (150, 78, 45, 0.7),
    (132, 88, 41, 0.5),
    (137, 82, 43, 0.6),
    (145, 86, 39, 0.4),
    (139, 84, 44, 0.7),
    (128, 91, 37, 0.5),
    (133, 77, 40, 0.6),
    (148, 83, 41, 0.5),
    (141, 89, 36, 0.4),
    (130, 80, 39, 0.6),
    (136, 85, 43, 0.5),
    (143, 76, 42, 0.7),
    (129, 92, 38, 0.4)
]

# Create a DataFrame from the new data
X_new = pd.DataFrame(new_data, columns=column_names)

# Load the trained model
model = joblib.load('machinemodel.pkl')

# Make predictions on the new data
predictions = model.predict(X_new)

# Print the predictions
print("Predictions for Maintenance Needed:")
for machine_id, prediction in enumerate(predictions, 1):
    print(f"Machine {machine_id}: {'Maintenance Needed' if prediction == 1 else 'No Maintenance Needed'}")
